integrationtest/spring/src/test/resources/persistence.xml

```diff
@@ -17,7 +17,7 @@
     </persistence-unit>
 
     <persistence-unit name="jbossjta" transaction-type="JTA">
-        <provider>org.hibernate.ejb.HibernatePersistence</provider>
+        <provider>org.hibernate.jpa.HibernatePersistenceProvider</provider>
```

Comment:
Seems like we could use the new class in `JBossTSIT.java`, too (see [here](https://github.com/hibernate/hibernate-search/blob/master/integrationtest/narayana/src/test/java/org/hibernate/search/test/integration/jbossjta/JBossTSIT.java#L88)).
Or was it on purpose that you skipped it?